### PR TITLE
Using RestartableJenkinsRule.then

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jenkinsci/jenkins-test-harness/pull/66.

@reviewbybees

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
🐝 once JTH is released

**Review by daniel-beck — APPROVED** [review]
Reviewer diff: https://github.com/jenkinsci/jenkins/pull/2959/files?w=1

**Comment by oleg-nenashev** [comment]
@reviewbybees done. Ready to go once the changes pass.
IMHO it requires an internal changelog entry since we pick major changes in JTH: https://github.com/jenkinsci/jenkins-test-harness#changelog . E.g. new Jetty version

**Comment by jglick** [comment]
> we pick major changes in JTH

So? They are not exposed to plugins.

**Comment by jglick** [comment]
Passing on Linux. The Windows nodes seem to be extremely slow. Need to investigate whether something in this PR is actually responsible.

File: test/pom.xml [hunk 1]
```diff
@@ -53,7 +53,7 @@ THE SOFTWARE.
     <dependency>
       <groupId>${project.groupId}</groupId>
       <artifactId>jenkins-test-harness</artifactId>
-      <version>2.20</version>
+      <version>2.24</version>
```
Comment (jglick) [inline test/pom.xml]: [Changelog](https://github.com/jenkinsci/jenkins-test-harness/#224-2017-aug-02)

I am wondering if https://github.com/jenkinsci/jenkins-test-harness/pull/63 is responsible for the [extremely slow Windows builds](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-2959/4/execution/node/34/log/?consoleFull) (2× time on Linux), or if this is just random bad luck. Other recent PRs seem to complete, though certainly more slowly than I would expect.

**Comment by oleg-nenashev** [comment]
I have put it on hold to let @jglick investigate the JTH issues.
I suffer from Windows build time in Remoting without Jetty, so I am not sure the  referenced PR is related

**Comment by oleg-nenashev** [comment]
Merge conflict

**Review by oleg-nenashev — APPROVED** [review]
🐝 if the test passes

File: test/src/test/java/hudson/cli/CLIActionTest.java [hunk 2]
```diff
@@ -255,6 +256,7 @@ public void encodingAndLocale() throws Exception {
         // -ssh mode does not pass client locale or encoding
     }
 
+    @Ignore("TODO JENKINS-46659 seems to be broken")
```
Comment (jglick) [inline test/src/test/java/hudson/cli/CLIActionTest.java]: [JENKINS-46659](https://issues.jenkins-ci.org/browse/JENKINS-46659) appears to have been the cause of the earlier failures, as well as those in #2754.

**Pull request merged**